library/src/main/java/com/datatorrent/lib/util/AppIdAwareStorageAgent.java

```diff
@@ -0,0 +1,37 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.datatorrent.lib.util;
+
+import com.datatorrent.api.StorageAgent;
+
+/**
+ * Interface to pass application id to storage agent
+ * 
+ *
+ */
+public interface AppIdAwareStorageAgent extends StorageAgent
```

Comment:
This is required as ApplicationAwareStorageAgent is not released in Apex core and build takes core dependency of release candidate but not develop branches.